x-pack/plugins/enterprise_search/server/routes/enterprise_search/engines.ts

```diff
@@ -88,13 +83,16 @@ export function registerEnginesRoutes({ log, router }: RouteDependencies) {
     },
     elasticsearchErrorHandler(log, async (context, request, response) => {
       const { client } = (await context.core).elasticsearch;
-      const engine =
-        await client.asCurrentUser.transport.request<EnterpriseSearchEngineUpsertResponse>({
-          body: { indices: request.body.indices },
-          method: 'PUT',
-          path: `/_application/search_application/${request.params.engine_name}`,
-          querystring: request.query,
-        });
+      const engine = (await client.asCurrentUser.searchApplication.put({
+        ...request.query,
+        name: request.params.engine_name,
+        search_application: {
+          indices: request.body.indices,
+          name: request.params.engine_name,
+          updated_at_millis: Date.now(),
```

Comment:
Take attention on this one. I had to do that according types, but ES client do that same if we don't pass anything